index.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+const Executor = require('screwdriver-executor-base');
+const path = require('path');
+const request = require('request');
+const jenkins = require('jenkins');
+const fs = require('fs');
+const async = require('async');
+
+class J5sExecutor extends Executor {
+
+    /**
+     * Constructor
+     * @method constructor
+     * @param  {Object} options           Configuration options
+     * @param  {Object} options.host      Jenkins hostname to make requests to
+     * @param  {Object} options.username  Jenkins username
+     * @param  {Object} options.password  Jenkins password/token
+     */
+    constructor(options) {
+        super();
+        this.host = options.host;
+        this.username = options.username;
+        this.password = options.password;
+        // need to pass port nubmer in the future
+        this.baseUrl = `http://${this.username}:${this.password}@${this.host}:8080`;
+        this.crumbUrl = this.baseUrl.concat('/crumbIssuer/api/json');
+    }
+
+    /**
+     * Get CSRF token crumb for Jenkins
+     * @method getCrumb
+     * @param  {Function} callback          Callback with error and crumb object
+     */
+    getCrumb(callback) {
+        const options = {
+            url: this.crumbUrl,
+            method: 'GET'
+        };
+
+        request(options, (error, response) => {
+            if (error) return callback(new Error(error.message));
+
+            if (response.statusCode !== 200) {
+                const msg = `Failed to get crumb: ${JSON.stringify(response.body)}`;
```

Comment:
Would it be useful to also include the status code in the message, or would that be too noisy?